src/overlays/actors/ovl_Eff_Zoraband/z_eff_zoraband.c

```diff
@@ -30,16 +30,71 @@ const ActorInit Eff_Zoraband_InitVars = {
     (ActorFunc)EffZoraband_Draw,
 };
 
-#endif
+void EffZoraband_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EffZoraband* this = THIS;
 
-extern UNK_TYPE D_06000F38;
+    Actor_SetScale(&this->actor, 1.0f);
+    this->actionFunc = EffZoraband_Mikau_FadeOut;
+    this->transparency = 0;
+    this->actor.shape.rot.z = 0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Eff_Zoraband/EffZoraband_Init.s")
+void EffZoraband_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Eff_Zoraband/EffZoraband_Destroy.s")
+void EffZoraband_Mikau_FadeOut(EffZoraband* this, GlobalContext* globalCtx) {
+    if (Cutscene_CheckActorAction(globalCtx, 0x20F) != 0) {
+        if ((EFFZORABAND_GET_F(&this->actor) + 2) ==
+            globalCtx->csCtx.actorActions[Cutscene_GetActorActionIndex(globalCtx, 0x20F)]->action) {
+            this->transpInit |= 2;
+        }
+        if (globalCtx->csCtx.actorActions[Cutscene_GetActorActionIndex(globalCtx, 0x20F)]->action == 7) {
+            this->actor.draw = NULL;
+        } else {
+            this->actor.draw = EffZoraband_Draw;
+        }
+    }
+    if ((this->actor.home.rot.z != 0) && (this->actor.draw != NULL)) {
+        func_800B9010(&this->actor, NA_SE_EV_UFO_LIGHT_BEAM - SFX_FLAG);
+    }
+    if ((this->transpInit & 2) != 0) {
```

Comment:
```suggestion
    if (this->transpInit & 2) {
```
Since they act like flags.